FireboltDotNetSdk.Tests/Integration/ConnectionTest.cs

```diff
@@ -58,7 +60,7 @@ public void FailedConnectTest(bool useDatabase, bool useEngine)
             {
                 connection.Open();
                 DbCommand command = connection.CreateCommand();
-                command.CommandText = "SELECT TOP 1 * FROM information_schema.tables";
+                command.CommandText = "SELECT 1";
                 command.ExecuteReader();
```

Comment:
I have changed the query although IMHO it is not needed. This test is expected to fail. It fails even with query `select 1`, so  there is not need to run more complicated query  to validate that connection fails. More complicated query even makes it worse:  it creates the illusion that it is required and makes sense while it does not. 

So, my solution is: add the query to the test parameters and run this test with both queries.

Reply:
That's not expected as `SELECT 1` is supposed to succeed when run on system engine. I think this might be an indication of a bug, please double check the logic.

Reply:
I did the following test. 
1. connected using UI, executed `select 1`, copied curl command - worked as expected
2. executed the same curl command in terminal - worked as expected
3. removed parameter `database` from URL and executed the command again - got error `invalid username or password for user`

Here is the output:
```
alexradzin@LenovoT14Firebolt:/tmp$ curl 'https://alex-test-general-purpose.firebolt.us-east-1.staging.firebolt.io/?database=alex_test&no_progress=true&bypass_queue=1&output_format=JSON_CompactLimited' \
  -H 'authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IlJrWTJRemMwUmpjMk0wWTFNalJGUXpKRU5qSTROMFpCUVRGQ056WXpSRUZCTmpJeE9USXhNQSJ9.eyJodHRwczovL0ZpcmVib2x0LmlvL3JvbGVzIjpbIkFDQ09VTlRfQURNSU4iXSwiaHR0cHM6Ly9GaXJlYm9sdC5pby91c2VyX25hbWUiOiJhbGV4YW5kZXIucmFkemluQGZpcmVib2x0LmlvIiwiaHR0cHM6Ly9GaXJlYm9sdC5pby9hY2NvdW50X25hbWUiOiJOL0EiLCJodHRwczovL2ZpcmVib2x0LmlvL2p3dC9jbGFpbXMiOnsicHJpbmNpcGFsIjoiL3VzZXJzLzdiNjNhYWRlLTBmYzUtNDRiNy1iYWM2LTYwZGQ2Mjc2NjRkMyIsImRlZmF1bHRfYWNjb3VudF9pZCI6IjI5N2RiNTI2LWYwYzEtNDZmMi05ZjY3LTZlOTE0MjkzMTQ3OSJ9LCJpc3MiOiJodHRwczovL2F1dGguc3RhZ2luZy5maXJlYm9sdC5pby8iLCJzdWIiOiJhdXRoMHw2NGI1MzFmZTkyZjU5ZjZiZTkyNTBhNmYiLCJhdWQiOiJodHRwczovL2FwaS5zdGFnaW5nLmZpcmVib2x0LmlvLyIsImlhdCI6MTcxMDM5OTIxMywiZXhwIjoxNzEwNDQyNDEzLCJhenAiOiIyWjNoaXlrUFBMOEpEbWtaYWVOV0owWmxjMnNjRWw3TSJ9.PGFXEJMnhXDusGV-thZ-bIOTlp1933mJPHGPwz98drDpHciCCn4FZSggAxiEPDIOOCSg161C8LPQerDU0c5BPcwDhQ2ANEHR4zxVVfLNf6oBqGOcklfzNTFFeWcgSb_rQdvhykXBCoctREbYn7smoggc3w3n9jH4zvvIhcAAngwoT9olCvRHUrgnt9MHQlvByN_hJ9C_4W7aBMnQM4LqteTmmvjYlfwIw0BlN3A5mXFhh7a3OSstt-UnTt6HOzEToi3DzL2yQTD-EVa7l0INQ6f1TdnvaEltoqBGvH0hZD8AvCw_7qgl4BhSmo3l5pnUkneRj2Iiad_3AemiHRykxQ' \
  --data-raw $'SELECT 1'
{
	"query":
	{
		"query_id": "17BC8F22187B5478"
	},
	"meta":
	[
		{
			"name": "?column?",
			"type": "int"
		}
	],

	"data":
	[
		[1]
	],

	"rows": 1,

	"statistics":
	{
		"elapsed": 0.001043523,
		"rows_read": 1,
		"bytes_read": 1,
		"time_before_execution": 0.000922308,
		"time_to_execute": 0.00012054,
		"scanned_bytes_cache": 0,
		"scanned_bytes_storage": 0
	}
}
alexradzin@LenovoT14Firebolt:/tmp$ 
alexradzin@LenovoT14Firebolt:/tmp$ 
alexradzin@LenovoT14Firebolt:/tmp$ curl 'https://alex-test-general-purpose.firebolt.us-east-1.staging.firebolt.io/?no_progress=true&bypass_queue=1&output_format=JSON_CompactLimited' \
  -H 'authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IlJrWTJRemMwUmpjMk0wWTFNalJGUXpKRU5qSTROMFpCUVRGQ056WXpSRUZCTmpJeE9USXhNQSJ9.eyJodHRwczovL0ZpcmVib2x0LmlvL3JvbGVzIjpbIkFDQ09VTlRfQURNSU4iXSwiaHR0cHM6Ly9GaXJlYm9sdC5pby91c2VyX25hbWUiOiJhbGV4YW5kZXIucmFkemluQGZpcmVib2x0LmlvIiwiaHR0cHM6Ly9GaXJlYm9sdC5pby9hY2NvdW50X25hbWUiOiJOL0EiLCJodHRwczovL2ZpcmVib2x0LmlvL2p3dC9jbGFpbXMiOnsicHJpbmNpcGFsIjoiL3VzZXJzLzdiNjNhYWRlLTBmYzUtNDRiNy1iYWM2LTYwZGQ2Mjc2NjRkMyIsImRlZmF1bHRfYWNjb3VudF9pZCI6IjI5N2RiNTI2LWYwYzEtNDZmMi05ZjY3LTZlOTE0MjkzMTQ3OSJ9LCJpc3MiOiJodHRwczovL2F1dGguc3RhZ2luZy5maXJlYm9sdC5pby8iLCJzdWIiOiJhdXRoMHw2NGI1MzFmZTkyZjU5ZjZiZTkyNTBhNmYiLCJhdWQiOiJodHRwczovL2FwaS5zdGFnaW5nLmZpcmVib2x0LmlvLyIsImlhdCI6MTcxMDM5OTIxMywiZXhwIjoxNzEwNDQyNDEzLCJhenAiOiIyWjNoaXlrUFBMOEpEbWtaYWVOV0owWmxjMnNjRWw3TSJ9.PGFXEJMnhXDusGV-thZ-bIOTlp1933mJPHGPwz98drDpHciCCn4FZSggAxiEPDIOOCSg161C8LPQerDU0c5BPcwDhQ2ANEHR4zxVVfLNf6oBqGOcklfzNTFFeWcgSb_rQdvhykXBCoctREbYn7smoggc3w3n9jH4zvvIhcAAngwoT9olCvRHUrgnt9MHQlvByN_hJ9C_4W7aBMnQM4LqteTmmvjYlfwIw0BlN3A5mXFhh7a3OSstt-UnTt6HOzEToi3DzL2yQTD-EVa7l0INQ6f1TdnvaEltoqBGvH0hZD8AvCw_7qgl4BhSmo3l5pnUkneRj2Iiad_3AemiHRykxQ' \
  --data-raw $'SELECT 1'
"10.0.2.27:2794": invalid username or password for user ""; query: "SELECT 1"
alexradzin@LenovoT14Firebolt:/tmp$ 

```

Reply:
Well, yes. Since you're connecting to a user engine URL it only works with a database defined in v1. Though it's weird it fails with "invalid username".
However, in your test case you're specifying useEngine = False, useDatabase = False. 
And please let's not post tokens on public discussion.

Reply:
So, is this discussion resolved?

Reply:
no?
We're testing that when neither database or engine is used a query that only fails when database is not specified will fail. `"SELECT TOP 1 * FROM information_schema.tables"` should fail, `SELECT 1` should not be failing. Please fix this test.

Reply:
I cannot fix test that fails. Did you see the curl?

Reply:
Did you see my reply explaining why this fails? You're using incorrect url.

Reply:
I have verified the things again and here is what I have found. 
When I call this function without engine (false, false) I do not have to run `select 1` to make this test to fail. It fails when it tries to find default engine URL by DB because the DB is not specified. 

However when engine is specified but DB is not the connection can can be opened but `select 1` fails. Here is call (without access token ;) )

Reply:
as discussed, let's make sure there's a test for v2 that ensures this query fails for when we connect to no database (with a system engine).

Reply:
We can easily connect to specific DB and run `select 1` using system engine. It works with UI as well. This scenario is validated by integration test also.